modules/Core/pages/register.php

```diff
@@ -220,6 +220,13 @@
                 },
             ]);
 
+            // Check if the ip they are trying to register with is banned
+            $ip = HttpUtils::getRemoteAddress();
+            if (DB::getInstance()->get('ip_bans', ['ip', $ip])->count()) {
+                Session::flash('home_error', $language->get('user', 'you_have_been_banned'));
```

Comment:
"Your IP has been banned, you cannot register." would be a more explicit message here